submissions/slack-status/src/background.js

```diff
@@ -0,0 +1,102 @@
+import { getSlackAuth } from "./slack.js";
+
+async function main() {
+  let auth;
+  try {
+    auth = await getSlackAuth();
+  } catch (e) {
+    console.error(e);
+    await chrome.tabs.create({ url: chrome.runtime.getURL("src/settings.html") });
+    return;
+  }
+
+  const { enabled, authorized } = await chrome.storage.local.get({ enabled: true, authorized: false });
+  if (!authorized) {
+    await chrome.tabs.create({ url: chrome.runtime.getURL("src/authorize.html") });
+  }
+
+  let lastTabId = -1;
+  while (true) {
+    let queryOptions = { active: true, lastFocusedWindow: true };
+    let [tab] = await chrome.tabs.query(queryOptions);
+    if (!tab) continue;
+    if (tab.id === lastTabId) continue;
+    lastTabId = tab.id || 0;
+    if (enabled) {
+      updateStatus(tab.id || 0, auth);
+    }
+    await delay(5000);
+  }
```

Comment:
I've modified the code to this and it's now very reactive. Immediately I switch tabs it changes on my Slack. I think you should get this to get your extension out the door. 

```js
  setInterval(async () => {
    let [tab] = await chrome.tabs.query({ active: true, lastFocusedWindow: true });
    if (enabled && tab.id !== lastTabId) updateStatus(tab.id || 0, auth);
  }, 2000);
```

On another note, it should be even better to change if there are changes to the url than the tab id. That's because someone can navigate to a different website while still being on the same tab.

Reply:
Thanks, updated my code!